### ISSUE TITLE
(CI) mirrorlist.centos.org is removed

### ISSUE THREAD

**Issue by oldherl** [dstndstn/astrometry.net#305]
As CentOS7 has ended support, the domain mirrorlist.centos.org is removed and can no longer be solved. This has caused the CI for CentOS7 to be failing.

It looks like that there's a workaround of changing it to vault.centos.org . Or we should just remove the CI for CentOS7 altogether.

**Comment by oldherl** [comment]
> Thanks for investigating! Any chance you would be interested in implementing this and sending a Pull Request? :)

But it's up to you to decide whether to continue the support for CentOS7, right?

**Comment by dstndstn** [comment]
I guess we can just drop it... that'll be an easy change.  Is there a newer RedHat variant we should support?

**Comment by oldherl** [comment]
I have no idea. I am personally an Arch Linux user. I bet a better alternative would be to add ubuntu 22.04 and rocky linux 9 support.

**Comment by dstndstn** [comment]
It does have Ubuntu 22.04 --
https://github.com/dstndstn/astrometry.net/blob/main/.circleci/config.yml#L115

Looks like we have rocky 8.5, so adding 9 probably wouldn't be too hard --
https://github.com/dstndstn/astrometry.net/blob/main/.circleci/config.yml#L387

I have never played with Arch so have no idea, but following the formula
for the closest distro would probably get you pretty far along.

But if you don't want to do this, that's fine too :)

**Comment by oldherl** [comment]
Actually I'm packaging this for Arch Linux in the [archlinuxcn] repo (https://github.com/archlinuxcn/repo/).
But I don't think Arch deserves a place in the CI.

The last comment should read 24.04 and I edited it. I don't know what was the problem for it that caused the disabling though.

Sure, I can write a PR that just adds a '#' in front of the centos7 line, but that just feels a little weird for my first contribution...

**Comment by dstndstn** [comment]
I just tried those changes, let's see what it does!

**Comment by oldherl** [comment]
Oh, please don't add centos8 as CentOS Linux 8 has reached End Of Life (EOL) on December 31st, 2021.
Yeah, it's even earlier than centos7.

**Comment by oldherl** [comment]
https://launchpad.net/ubuntu/noble/amd64/libpython3.11-dev
python 3.11 has been removed for ubuntu 24.04, please use python 3.12 instead

**Comment by dstndstn** [comment]
Thanks for the tips!  Is there a live centos?

**Comment by oldherl** [comment]
They switched to "Stream", which is a rolling release style just like Arch. So I'm not sure if that is worth a CI now.

**Comment by oldherl** [comment]
Python 3.12 is now by default preventing the use of `pip install` outside a venv. I would prefer to use apt to install the python libraries `apt install python3-fitsio python3-numpy`.

**Comment by oldherl** [comment]
```
depends=('bzip2' 'cairo' 'libpng' 'libjpeg-turbo' 'python-astropy' 'netpbm' 'cfitsio' 'gsl' 'swig')
makedepends=('python-setuptools')
```
This is the dependency list I use on Arch. Maybe you could find the equivalents on Ubuntu. Good luck :)

**Comment by dstndstn** [comment]
Looks like it's working, shall we close this ticket?

**Comment by oldherl** [comment]
Fixed in bf2826d
